axelor-production/src/main/java/com/axelor/apps/production/service/MrpServiceProductionImpl.java

```diff
@@ -443,60 +461,78 @@ protected void createProposalMrpLine(
       throws AxelorException {
 
     Company company = mrp.getStockLocation().getCompany();
-    BillOfMaterial defaultBillOfMaterial = billOfMaterialService.getDefaultBOM(product, company);
+
+    MrpLine originalMrpLine =
+        mrpLineOriginList.stream().findAny().map(MrpLineOrigin::getMrpLine).orElse(null);
+
+    BillOfMaterial billOfMaterial;
```

Comment:
```java
 if (originalMrpLine != null) {
      if (originalMrpLine.getBillOfMaterial() != null) {
        billOfMaterial = originalMrpLine.getBillOfMaterial();
      } else {
        billOfMaterial =
            billOfMaterialService.getEligibleBillOfMaterialOfProductInMrpLine(
                originalMrpLine, product);
      }
    } else {
      billOfMaterial = billOfMaterialService.getDefaultBOM(product, company);
    }
```  
Isn't this equivalent of just calling` getEligibleBillOfMaterialOfProductInMrpLine `?